executor/brie.go

```diff
@@ -273,29 +431,167 @@ func (b *executorBuilder) buildBRIE(s *ast.BRIEStmt, schema *expression.Schema)
+		// global config for lightning as a long-term running server that serve many tasks
+		importGlobalCfg = importcfg.NewGlobalConfig()
+		importTaskCfg   = importcfg.NewConfig()
+	)
+	// set proper values to embedded lightning
+	importGlobalCfg.App.StatusAddr = ":8289"
+	importGlobalCfg.App.Level = tidbCfg.Log.Level
+	importGlobalCfg.Security.CAPath = tidbCfg.Security.ClusterSSLCA
+	importGlobalCfg.Security.CertPath = tidbCfg.Security.ClusterSSLCert
+	importGlobalCfg.Security.KeyPath = tidbCfg.Security.ClusterSSLKey
+	importGlobalCfg.TiDB.Port = int(tidbCfg.Port)
+	importGlobalCfg.TiDB.PdAddr = strings.Split(tidbCfg.Path, ",")[0]
+
+	// set Port and PdAddr to avoid lightning query TiDB HTTP API
+	importTaskCfg.TiDB.Port = int(tidbCfg.Port)
+	importTaskCfg.TiDB.PdAddr = strings.Split(tidbCfg.Path, ",")[0]
+	// StatusPort is not used, but we could pass it to improve robust
+	importTaskCfg.TiDB.StatusPort = int(tidbCfg.Status.StatusPort)
+	importTaskCfg.TikvImporter.Backend = importcfg.BackendLocal
+	importTaskCfg.TikvImporter.SortedKVDir = filepath.Join(tidbCfg.TempStoragePath, defaultImportID)
+	importTaskCfg.Checkpoint.Schema = defaultImportID
+	importTaskCfg.Checkpoint.Driver = importcfg.CheckpointDriverMySQL
+	importTaskCfg.Mydumper.CSV.Header = false // TODO(lance6716): address this behaviour to user
+
+	// handle s.Storage:
+	// IMPORT ... FROM **s.Storage**
+	storageURL, err := url.Parse(s.Storage)
+	if err != nil {
+		b.err = errors.Annotate(err, "invalid destination URL")
 		return nil
 	}
+	importTaskCfg.Mydumper.SourceDir = storageURL.String()
+	e.info.storage = storageURL.String()
+
+	// handle s.Tables:
+	// IMPORT **s.Tables** FROM ...
+	switch {
+	case len(s.Tables) != 0:
+		tbls := make([]string, 0, len(s.Tables))
+		for _, tbl := range s.Tables {
+			if tbl.Schema.L == "" {
+				b.err = errors.Errorf("please specify schema for %s in IMPORT", tbl.Name.O)
+				return nil
+			}
+			tbls = append(tbls, fmt.Sprintf("%s.%s", tbl.Schema, tbl.Name))
+		}
+		importTaskCfg.Mydumper.Filter = tbls
+	case len(s.Schemas) != 0:
+		dbs := make([]string, 0, len(s.Schemas))
+		for _, db := range s.Schemas {
+			dbs = append(dbs, fmt.Sprintf("%s.*", db))
+		}
+		importTaskCfg.Mydumper.Filter = dbs
+	default:
+		// no filter means pass all
+	}
+
+	// handle options
+	for _, opt := range s.Options {
+		switch opt.Tp {
+		case ast.BRIEOptionSkipSchemaFiles:
+			importTaskCfg.Mydumper.NoSchema = opt.UintValue != 0
+		case ast.BRIEOptionStrictFormat:
+			importTaskCfg.Mydumper.StrictFormat = opt.UintValue != 0
+		case ast.BRIEOptionCSVSeparator:
+			importTaskCfg.Mydumper.CSV.Separator = opt.StrValue
+		case ast.BRIEOptionCSVDelimiter:
+			importTaskCfg.Mydumper.CSV.Delimiter = opt.StrValue
+		case ast.BRIEOptionCSVHeader:
+			if opt.UintValue == ast.BRIECSVHeaderIsColumns {
+				importTaskCfg.Mydumper.CSV.Header = true
+			} else {
+				b.err = errors.Errorf("CSV_HEADER only support FIELDS or COLUMNS to indicate it has header")
```

Comment:
error code

Reply:
some of error code was wrapped using `handleBRIEError`. I'll update build time error code later

Reply:
updated in https://github.com/pingcap/tidb/pull/21195/commits/7d41ff61d3de3190de0e322c57143ef2e261ef8b. and some functions called by `Next` are returning error directly and `Next` wrap them using `handleBRIEError`